lib/fluent/plugin/out_elasticsearch_data_stream.rb

```diff
@@ -41,54 +41,65 @@ def configure(conf)
         raise Fluent::ConfigError, "'elasticsearch/xpack'' is required for <@elasticsearch_data_stream>."
       end
 
-      begin
-        @client = client
-        create_ilm_policy
-        create_index_template
-        create_data_stream
-      rescue => e
-        raise Fluent::ConfigError, "Failed to create data stream: <#{@data_stream_name}> #{e.message}"
+      @client = client
+      if placeholder?(:data_stream_name_placeholder, @data_stream_name)
+        @use_placeholder = true
+        @data_stream_names = []
+      else
+        begin
+          @data_stream_names = [@data_stream_name]
+          create_ilm_policy(@data_stream_name)
+          create_index_template(@data_stream_name)
+          create_data_stream(@data_stream_name)
+        rescue => e
+          raise Fluent::ConfigError, "Failed to create data stream: <#{@data_stream_name}> #{e.message}"
+        end
       end
     end
 
-    def create_ilm_policy
+    def create_ilm_policy(name)
       params = {
-        policy_id: "#{@data_stream_name}_policy",
+        policy_id: "#{name}_policy",
         body: File.read(File.join(File.dirname(__FILE__), "default-ilm-policy.json"))
       }
       @client.xpack.ilm.put_policy(params)
     end
 
-    def create_index_template
+    def create_index_template(name)
       body = {
-        "index_patterns" => ["#{@data_stream_name}*"],
+        "index_patterns" => ["#{name}*"],
         "data_stream" => {},
         "template" => {
           "settings" => {
-            "index.lifecycle.name" => "#{@data_stream_name}_policy"
+            "index.lifecycle.name" => "#{name}_policy"
           }
         }
       }
       params = {
-        name: @data_stream_name,
+        name: name,
         body: body
       }
       @client.indices.put_index_template(params)
     end
 
-    def create_data_stream
+    def data_stream_exist?(name)
       params = {
-        "name": @data_stream_name,
+        "name": name
       }
       begin
         response = @client.indices.get_data_stream(params)
-        unless response.is_a?(Elasticsearch::Transport::Transport::Errors::NotFound)
-          log.info "Specified data stream exists: <#{@data_stream_name}>"
-          return
-        end
+        return (not response.is_a?(Elasticsearch::Transport::Transport::Errors::NotFound))
       rescue Elasticsearch::Transport::Transport::Errors::NotFound => e
         log.info "Specified data stream does not exist. Will be created: <#{e}>"
+        return false
       end
+    end
+
+    def create_data_stream(name)
+      params = {
+        "name": name
+      }
+      return if data_stream_exist?(name)
```

Comment:
We can use this line in the first line on `#create_data_stream` to prevent params object creation.

Reply:
Sure,